libs/@hashintel/ds-components/src/components/Button/button.figma.tsx

```diff
@@ -0,0 +1,46 @@
+import figma from "@figma/code-connect";
+
+import { Button } from "./button";
+
+/**
+ * -- This file was auto-generated by Code Connect --
+ * `props` includes a mapping from your code props to Figma properties.
+ * You should check this is correct, and update the `example` function
+ * to return the code example you'd like to see in Figma
+ */
+
+figma.connect(
+  Button,
+  "https://www.figma.com/design/WmnosvOvi4Blw0HK0jh1mG/Design-System?node-id=198%3A41491",
+  {
+    props: {
+      // These props were automatically mapped based on your linked code:
+      disabled: figma.enum("_state", {
+        disabled: true,
+      }),
+      // No matching props could be found for these Figma properties:
+      // "children": figma.string('children'),
+      // "iconLeft": figma.instance('iconLeft'),
+      // "showIconLeft": figma.boolean('_showIconLeft'),
+      // "iconRight": figma.instance('iconRight'),
+      // "showIconRight": figma.boolean('_showIconRight'),
+      // "variant": figma.enum('variant', {
+      //   "primary": "primary",
+      //   "secondary": "secondary",
+      //   "ghost": "ghost"
+      // }),
+      // "size": figma.enum('size', {
+      //   "xs (24px)": "xs--24px-",
+      //   "sm (28px)": "sm--28px-",
+      //   "md (32px)": "md--32px-",
+      //   "lg (40px)": "lg--40px-"
+      // }),
+      // "colorScheme": figma.enum('colorScheme', {
+      //   "neutral": "neutral",
+      //   "brand": "brand",
+      //   "critical": "critical"
+      // })
+    },
+    example: (props) => <Button disabled={props.disabled} />,
+  },
```

Comment:
Props are still badly defined at this step.
We still need to:

- Define correct props in React components
- Verify properties in the Figma

This will become better with iteration.